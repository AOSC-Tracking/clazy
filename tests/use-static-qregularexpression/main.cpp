#include <QtCore/QObject>
#include <QtCore/QString>
#include <QtCore/QRegularExpression>

void test()
{
    if (QString("abc").contains(QRegularExpression("[a-z]"))) { // Warn
    }

    QString h = "hello";
    auto c = h.indexOf(QRegularExpression("[hel]"), 0);  // Warn

    QRegularExpression e("[a-z]");
    auto r = h.indexOf(e); // Warn

    static const QRegularExpression staticRegex("[a-z]");
    auto nr = h.indexOf(staticRegex); // Ok

    bool empty = QString().isEmpty();

    h.lastIndexOf(QRegularExpression("[hel]")); // Warn
    h.lastIndexOf(e); // Warn
    h.lastIndexOf(staticRegex); // Ok

    h.count(QRegularExpression("[hel]"));  // Warn
    h.count(e);  // Warn
    h.count(staticRegex); // Ok

    h.replace(QRegularExpression("[hel]"), QString()); // Warn
    h.replace(e, QString()); // Warn
    h.replace(staticRegex, QString()); // Ok

    h.remove(QRegularExpression("[hel]")); // Warn
    h.remove(e); // Warn
    h.remove(staticRegex); // Ok

    h.section(QRegularExpression("[hel]"), 0); // Warn
    h.section(e, 0); // Warn
    h.section(staticRegex, 0); // Ok

    h.split(QRegularExpression("[hel]"));  // Warn
    h.split(e);  // Warn
    h.split(staticRegex); // Ok

    h.splitRef(QRegularExpression("[hel]"));  // Warn
    h.splitRef(e);  // Warn
    h.splitRef(staticRegex); // Ok

    QStringList strList;

    strList.indexOf(e); // Warn
    strList.indexOf(staticRegex);

    strList.lastIndexOf(e); // Warn
    strList.lastIndexOf(staticRegex); // Ok
}
