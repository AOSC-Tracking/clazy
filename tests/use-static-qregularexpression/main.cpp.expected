use-static-qregularexpression/main.cpp:6:33: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:10:24: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:13:24: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:20:19: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:21:19: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:24:13: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:25:13: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:28:15: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:29:15: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:32:14: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:33:14: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:36:15: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:37:15: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:40:13: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:41:13: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:46:21: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:49:25: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:53:16: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:57:20: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:89:19: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:90:19: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:92:15: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
use-static-qregularexpression/main.cpp:93:15: warning: Don't create temporary QRegularExpression objects. Use a static QRegularExpression object instead [-Wclazy-use-static-qregularexpression]
